java-client/src/main/java/org/opensearch/client/transport/aws/AwsV2OpenSearchTransport.java

```diff
@@ -0,0 +1,427 @@
+                }));
+    }
+
+    @Override
+    public JsonpMapper jsonpMapper() {
+        return mapper;
+    }
+
+    @Override
+    public TransportOptions options() {
+        return transportOptions;
+    }
+
+    @Override
+    public void close() {
+    }
+
+    private <RequestT> byte[] prepareRequestBody(
+            RequestT request,
+            Endpoint<RequestT, ?, ?> endpoint
+    ) {
+        if (endpoint.hasRequestBody()) {
+            // Request has a body and must implement JsonpSerializable or NdJsonpSerializable
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+            if (request instanceof NdJsonpSerializable) {
+                writeNdJson((NdJsonpSerializable) request, baos);
+            } else {
+                JsonGenerator generator = mapper.jsonProvider().createGenerator(baos);
+                mapper.serialize(request, generator);
+                generator.close();
+            }
+            return baos.toByteArray();
+        }
+        return null;
+    }
+
+    private <RequestT> SdkHttpFullRequest prepareRequest(
+            RequestT request,
+            Endpoint<RequestT, ?, ?> endpoint,
+            @Nullable TransportOptions options,
+            @Nullable byte[] body
+    ) {
+        SdkHttpFullRequest.Builder req = SdkHttpFullRequest.builder()
+                .method(SdkHttpMethod.fromValue(endpoint.method(request)));
+
+        StringBuilder url = new StringBuilder();
+        url.append("https://").append(host);
+        String path = endpoint.requestUrl(request);
+        if (!path.startsWith("/")) {
+            url.append('/');
+        }
+        url.append(path);
+        Map<String, String> params = endpoint.queryParameters(request);
+        if (params != null && !params.isEmpty()) {
+            char sep = '?';
+            for (var ent : params.entrySet()) {
+                url.append(sep).append(ent.getKey()).append('=');
+                url.append(URLEncoder.encode(ent.getValue(), StandardCharsets.UTF_8));
+                sep = '&';
+            }
+        }
+        applyOptionsParams(url, transportOptions);
+        applyOptionsParams(url, options);
+        try {
+            req.uri(new URI(url.toString()));
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException("Invalid request URI: " + url.toString());
+        }
+        applyOptionsHeaders(req, transportOptions);
+        applyOptionsHeaders(req, options);
+        if (endpoint.hasRequestBody() && body != null) {
+            req.putHeader("Content-Type", "application/json");
+            req.putHeader("Content-Length", String.valueOf(body.length));
+            req.contentStreamProvider(() -> new ByteArrayInputStream(body));
+        }
+
+        Aws4SignerParams signerParams = Aws4SignerParams.builder()
+                .awsCredentials(credentials.resolveCredentials())
+                .signingName("es")
+                .signingRegion(Region.US_EAST_1)
```

Comment:
We need to extract the signing region into a parameter here.

Reply:
Oops. yes, of course